### PR TITLE
Feature / 30 table block styling

### PR DESCRIPTION
**Pull request by sbrody**
Legacy table styling is removed in favour of GOV.UK styles. This should apply to new tables added via the block editor and existing ones displayed via the classic editor block.

The markup does not match the [GOV.UK design system](https://design-system.service.gov.uk/components/table/) most notably with the lack of a `<caption>` element. WordPress uses a `<figcaption>` instead, which is styled in this PR to look like the GOV.UK caption.

Note: there are a few configuration options for GOV.UK tables but they are not included here for reasons of simplicity.

Updated image:
image

### CODE REVIEW

File: assets/css/components/tables.scss
Comment (matthewpassmore) [inline assets/css/components/tables.scss:1]: There are duplicate styles in `global-styles.scss`?

File: assets/css/components/tables.scss [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+table thead {
+	border: none !important;
+}
+
+table th, table td {
+	border-top: none !important;
+	border-left: none !important;
+	border-right: none !important;
```
Comment (matthewpassmore) [inline assets/css/components/tables.scss:22]: **question:** can the use of `!important` be circumvented with a slight bump in specificity e.g. prepending a class to bump by by `10`?

**Review by matthewpassmore — COMMENTED** [review]
**suggestion:** style all first column data as bold, to align with GOV.UK tables. Though the markup is different (GOV.UK uses `<th>`), we could do something along the lines of:

```css
tr > td:first-child { @include govuk-typography-weight-bold; }
```

File: assets/css/components/tables.scss
Comment (sbrody) [reply assets/css/components/tables.scss:1]: Have removed now

File: assets/css/components/tables.scss (hunk 1 above)
Comment (sbrody) [reply assets/css/components/tables.scss]: Have added main-content class, so both table block and legacy tables are included

**Comment by sbrody** [comment]
> **suggestion:** style all first column data as bold, to align with GOV.UK tables. Though the markup is different (GOV.UK uses `<th>`), we could do something along the lines of:
> 
> ```css
> tr > td:first-child { @include govuk-typography-weight-bold; }
> ```

Have added that in

File: assets/css/components/tables.scss (hunk 1 above)
Comment (matthewpassmore) [reply assets/css/components/tables.scss]: @sbrody This change prevents visual parity with the Block Editor, as `.main-content` isn't a thing in that context.

**Review by matthewpassmore — CHANGES_REQUESTED** [review]
Visual parity between frontend and Block Editor needs to be looked at, thanks.

**Review by matthewpassmore — APPROVED** [review]
Looks good to me 👍

**Pull request merged**